### DISCUSSION TITLE
PyInstaller/PyQt6 app built on GH Actions does not run on macOS 10.15

### DISCUSSION THREAD

**Discussion by justvanrossum** [pyinstaller/pyinstaller#7563] [Help]
Apologies if this is more of a PyQt6-question.

My PyQt6 app runs nicely on macOS 10.15, when built (with PyInstaller) on said 10.15 system.

When I build the app on GH Actions (which uses macOS 12), using the exact same version of Python (3.10.9, explicitly downloaded from python.org) and the exact same dependencies (latest PyInstaller + PyQt6), the app does not run on macOS 10.15, and outputs this error:

> Sorry, "Fontra Pak" cannot be run on this version of macOS. Qt requires macOS 11.0.0 or later, you have macOS 10.15.7.

The app is x86-only, not universal.

I am very confused, as the app built on GH Actions *used* to work on macOS 10.15, not that long ago (at least as recently as 17 March 2023). I tried to downgrade PyQt and PyInstaller but the error does not go away. I tried to downgrade the version of XCode on the GH runner, to no avail either. What else could have changed?

I may be missing something obvious, but right now I stumped. Does anyone here have an idea?

**Comment by bwoodsend** [comment]
For starters, the python environments provided by `actions/setup-python` are built to support macOS >=11 only. If anything from brew is part of your application, that'll all be built for macOS 12.

If you intend to support older versions of mac, you have to make sure that every binary file you include is compiled with a compatible deployment target.

**Reply by justvanrossum** [reply]
Thanks, I will check. That's a good clue: maybe some wheels are implicitly built on the runner, instead of using a prebuilt one from PyPI.

I don't use brew, and I don't use actions/setup-python, but Python 3.10.9 directly downloaded from python.org.

**Comment by justvanrossum** [comment]
I'm a big step further: I can now reproduce the problem locally, so this rules out a GH-Actions specific problem. It is probably some PyPI dependency that I haven't got properly pinned.

I'll close the discussion, but will comment once I've found the culprit.

**Comment by justvanrossum** [comment]
It turns out the latest PyQt6-Qt6, a dependency of PyQt6 does not run on macOS 10.15, despite the x86 wheel being advertised as such. I'll try to file an issue with Qt.

It had nothing to do with PyInstaller. Sorry for the noise, and thanks for the help.
